build/gen_stub.php

```diff
@@ -2918,6 +2918,100 @@ class PropertyInfo extends VariableLike
     public bool $isDocReadonly;
     public bool $isVirtual;
 
+    // Map possible variable names to the known string constant, see
+    // ZEND_KNOWN_STRINGS
+    private const PHP_80_KNOWN = [
```

Comment:
Do you have any rough estimate how much performance/memory this optimization saves? Since property registration happens during `MINIT`, it only affects startup time (so it's less important for the regular use-case), and the memory reduction may be a tiny bit (?). I'm asking these because the double bookkeeping of known strings may increase the (already big) cognitive load, so we should carefully evaluate the advantages/disadvantages.

Reply:
No idea, sorry - just another one of the ways I saw to simplify the registration, since I came across
https://github.com/php/php-src/blob/db9a7aaf05c363f0345a90d6d8e0df3fcd4831bd/Zend/zend_enum.c#L474
where the enum property `name` is done with a known string, but https://github.com/php/php-src/blob/db9a7aaf05c363f0345a90d6d8e0df3fcd4831bd/ext/dom/php_dom_arginfo.h#L2013-L2015 where a property be the same name is done with `zend_string_init()` and figured if the hand-written registration can do this, why not the auto-generated?

Reply:
But since a) the arginfo files are automatically generated, and b) there is no problem with not having some of the known strings added to the auto generation, since it is obviously still fine to use the manual strings, and c) there is presumably some non-zero memory reduction, I figured it would be helpful to implement this

Reply:
Yeah, to be clear, I like the idea, it's clever, however I am not a fan of updating/adding known strings at two places instead of just one. I'd say that let's wait for the opinion of other reviewers!

Reply:
I like the PR but indeed having to manage the known strings in 2 places is annoying and will be forgotten for sure at some point in the future. I think it's possible to do some ad hoc solution in the stub generator that searches the known strings in the C file, this can be done in an ad hoc way imo, no need to do any fancy parsing, just some pattern matching should do.

Reply:
> will be forgotten for sure at some point in the future

There isn't any harm in not updating the know strings in the arginfo generator

> searches the known strings in the C file

That would only be possible for the latest version of the strings though, it wouldn't indicate the version they were added. What about just adding a note to https://github.com/php/php-src/blob/f17c2203883ddf53adfcb33d85523d11429729ab/Zend/zend_string.h#L563 to also update the generator script?

Reply:
A note is better than nothing, but it wouldn't be the first time a note gets ignored. Indeed it would not be harmful, it would just be nicer if it were automatic, but I see the problem in trying to implement that.

Reply:
Note added